File: Chapter2/TwoSix.java
// checks if a linked list is a palindrome
public class TwoSix {

    Boolean check(LinkedListNode root)
    {
        LinkedListNode two = new LinkedListNode(root.getValue());
        LinkedListNode pointer = root;
        int count = 1;
        while(pointer.getNext() != null)
        {
            pointer = pointer.getNext();
            LinkedListNode temp = new LinkedListNode(pointer.getValue());
            temp.setNext(two);
            two = temp;
            count++;
        }
        for (int i = 0; i < count / 2; i++)
        {
            if(root.getValue() != two.getValue())
            {
                return false;
            }
        }
        return true;
    }
    
}

File: Chapter2/two-six.py
# checks if a linked list is a palindrome given the root node
def check(root):
    ll2 = node(root.value)
    pointer = root
    count = 1
    while(root.next != None):
        pointer = pointer.next
        temp = node(pointer.value)
        temp.next = ll2
        ll2 = temp
        count += 1
    for i in range(count//2):
        if(ll2.value != root.value):
            return False
    return True
    

